subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/notes/ReleaseNotesPlugin.java

```diff
@@ -51,46 +53,76 @@ public void apply(final Project project) {
 
-                boolean previewMode = project.hasProperty(PREVIEW_PROJECT_PROPERTY);
-                t.setPreviewMode(previewMode);
+            boolean previewMode = project.hasProperty(PREVIEW_PROJECT_PROPERTY);
+            task.setPreviewMode(previewMode);
 
-                if (!previewMode) {
-                    File releaseNotesFile = project.file(conf.getReleaseNotes().getFile());
-                    GitPlugin.registerChangesForCommitIfApplied(
-                        singletonList(releaseNotesFile), "release notes updated", t);
-                    t.getOutputs().file(releaseNotesFile);
-                }
+            if (!previewMode) {
+                File releaseNotesFile = project.file(conf.getReleaseNotes().getFile());
+                GitPlugin.registerChangesForCommitIfApplied(
+                    singletonList(releaseNotesFile), "release notes updated", task);
+                task.getOutputs().file(releaseNotesFile);
             }
         });
+
+        UpdateReleaseNotesOnGitHubTask updateReleaseNotesOnGitHubTask = TaskMaker.task(project, UPDATE_NOTES_ON_GITHUB_TASK, UpdateReleaseNotesOnGitHubTask.class, task -> {
+            task.setDescription("Updates release notes on GitHub releases page. Run with '-Ppreview' if you only want to see the preview.");
+
+            configureDetailedNotes(task, releaseNotesFetcher, project, conf, contributorsFetcher);
+
+            boolean previewMode = project.hasProperty(PREVIEW_PROJECT_PROPERTY);
+            task.setPreviewMode(previewMode);
+
+            LazyConfiguration.lazyConfiguration(task, () -> {
+                task.setGitHubWriteToken(conf.getGitHub().getWriteAuthToken());
+            });
+        });
+
+        updateReleaseNotesOnGitHubTask.setGitHubApiUrl(conf.getGitHub().getApiUrl());
+        updateReleaseNotesOnGitHubTask.setUpstreamRepositoryName(conf.getGitHub().getRepository());
+        updateReleaseNotesOnGitHubTask.setDryRun(conf.isDryRun());
+
+        UpdateReleaseNotesOnGitHubCleanupTask updateReleaseNotesOnGitHubCleanupTask = TaskMaker.task(project, UPDATE_NOTES_ON_GITHUB_CLEANUP_TASK, UpdateReleaseNotesOnGitHubCleanupTask.class, task -> {
+            task.setDescription("Remove release notes from GitHub release page created by updateReleaseNotesOnGitHub task.");
+
+            configureDetailedNotes(task, releaseNotesFetcher, project, conf, contributorsFetcher);
+
+            boolean previewMode = project.hasProperty(PREVIEW_PROJECT_PROPERTY);
+            task.setPreviewMode(previewMode);
+
+            LazyConfiguration.lazyConfiguration(task, () -> {
+                task.setGitHubWriteToken(conf.getGitHub().getWriteAuthToken());
+            });
+        });
+
+        updateReleaseNotesOnGitHubCleanupTask.setGitHubApiUrl(conf.getGitHub().getApiUrl());
+        updateReleaseNotesOnGitHubCleanupTask.setUpstreamRepositoryName(conf.getGitHub().getRepository());
+        updateReleaseNotesOnGitHubCleanupTask.setPublicationRepository("aa");
```

Comment:
hmm, this one looks suspicious.

Reply:
Good catch! This line will be deleted by me in the next PR. The proper value is set in BintrayReleasePlugin.java:65